File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s) {
  sArr = s.split("");
  const heap = [sArr.pop()];

  while (sArr.length > 0) {
      temp = sArr.pop()
      if(heap[heap.length-1] === temp) heap.pop()
      else heap.push(temp)
    //heap.push(sArr.pop());
    //const len = heap.length;
    //if (heap[len - 1] === heap[len - 2]) heap.splice(-2, 2);
  }

  return heap.length > 0 ? 0 : 1;
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
class Solution
{
    public int solution(String s)
    {
        char[] str = new char[s.length()+1];
        int k = 1;
        int answer = -1;
        
        for(int i = 0;i<s.length();i++){
            str[k]=s.charAt(i);
            if(str[k]==str[k-1]) k-=2;
            k++;
        }

        if(k==1) answer=1;
        else answer=0;

        return answer;
    }
}